service/src/test/java/uk/gov/hmcts/reform/fpl/service/removeorder/RemoveOrderServiceTest.java

```diff
@@ -0,0 +1,207 @@
+            .value(DynamicListElement.EMPTY)
+            .listItems(List.of(
+                buildListElement(generatedOrders.get(0).getId(), "order 1 - 15 June 2020"),
+                buildListElement(generatedOrders.get(1).getId(), "order 2 - 16 July 2020")
+            ))
+            .build();
+
+        assertThat(listOfOrders).isEqualTo(expectedList);
+    }
+
+    @Test
+    void shouldMakeDynamicListOfMixedOrderTypes() {
+        List<Element<GeneratedOrder>> generatedOrders = List.of(
+            element(buildOrder(BLANK_ORDER, "order 1", "15 June 2020")),
+            element(buildOrder(CARE_ORDER, "order 2", "16 July 2020")),
+            element(buildOrder(EMERGENCY_PROTECTION_ORDER, "order 3", "17 August 2020")),
+            element(buildOrder(SUPERVISION_ORDER, "order 4", "18 September 2020"))
+        );
+
+        List<Element<CaseManagementOrder>> caseManagementOrders = buildCaseManagementOrders();
+
+        CaseData caseData = CaseData.builder()
+            .orderCollection(generatedOrders)
+            .sealedCMOs(caseManagementOrders)
+            .build();
+
+        DynamicList listOfOrders = underTest.buildDynamicListOfOrders(caseData);
+
+        DynamicList expectedList = DynamicList.builder()
+            .value(DynamicListElement.EMPTY)
+            .listItems(List.of(
+                buildListElement(generatedOrders.get(0).getId(), "order 1 - 15 June 2020"),
+                buildListElement(generatedOrders.get(1).getId(), "order 2 - 16 July 2020"),
+                buildListElement(generatedOrders.get(2).getId(), "order 3 - 17 August 2020"),
+                buildListElement(generatedOrders.get(3).getId(), "order 4 - 18 September 2020"),
+                buildListElement(caseManagementOrders.get(0).getId(), format("Case management order - %s",
+                    formatLocalDateToString(NOW, "d MMMM yyyy")))))
+            .build();
+
+        assertThat(listOfOrders).isEqualTo(expectedList);
+    }
+
+    @Test
+    void shouldPopulateCaseFieldsFromCaseWithMatchedAction() {
+        when(orderRemovalActions.getActions()).thenReturn(List.of(
+            orderRemovalAction,
+            anotherOrderRemovalAction
+        ));
+        when(orderRemovalAction.isAccepted(removableOrder)).thenReturn(true);
+
+        underTest.populateSelectedOrderFields(caseData, data, REMOVED_UUID, removableOrder);
+
+        verify(orderRemovalAction).populateCaseFields(caseData, data, REMOVED_UUID, removableOrder);
+        verifyNoInteractions(anotherOrderRemovalAction);
+    }
+
+    @Test
+    void shouldThrowExceptionIfActionNotMatchedPopulateCaseFieldsFromCase() {
+        when(orderRemovalActions.getActions()).thenReturn(List.of(
+            orderRemovalAction,
+            anotherOrderRemovalAction
+        ));
+        when(orderRemovalAction.isAccepted(removableOrder)).thenReturn(false);
+        when(anotherOrderRemovalAction.isAccepted(removableOrder)).thenReturn(false);
+
```

Comment:
is it worth having a test scenario where one action returns true, the other returns false?

Reply:
Discussed